snuba/migrations/table_engines.py

```diff
@@ -79,8 +82,26 @@ def __init__(
         super().__init__(order_by, partition_by, sample_by, ttl, settings)
         self.__version_column = version_column
 
-    def _get_engine_type(self, single_node: bool, table_name: str) -> str:
-        if single_node:
+    def _get_engine_type(self, storage_set_key: StorageSetKey, table_name: str) -> str:
```

Comment:
I am not sure letting the StorageSetKey type creep so low in the stack would be a good idea.
It would prevent us from removing a storage set from the system ever.
I wonder whether it would be a better idea to make the migration (or even the migration runner resolve the cluster and pass that object down to the operations). The less the operations depend on types that represent concepts in the dataset datamodel, the more future proof the migration itself is.

Reply:
I'd be happy to switch this out for the resolved cluster since that's all the table engines really need and forward that here instead. But I highly doubt this is what prevents us from removing a storage set from the system since it's just a parameter passed directly from the operation which is provided there by the user (it has to be since storage set is the only reference to a cluster we have). I think what actually makes removing a storage set hard is that it potentially involves needing to copy large amounts of data from one machine to another.

Reply:
Updated to cluster here

Reply:
I think switching storage_set to be a string instead of enum in our migrations sounds the most straightforward option. When we encounter a storage set that's not in our enum anymore we'd have to assume it's a deprecated one and skip that operation and move onto the next. We'd have to give a lot of warnings to the user to make sure it's not a typo since that would be the other possibility. Otherwise we'd have to maintain a list of deprecated storage set keys.